lib/iris/util.py

```diff
@@ -1589,3 +1589,35 @@ def demote_dim_coord_to_aux_coord(cube, name_or_coord):
     cube.remove_coord(dim_coord)
 
     cube.add_aux_coord(dim_coord, coord_dim)
+
+
+def _num2date_to_nearest_second(time_value, units):
```

Comment:
I think that this function may prove to be useful for users experiencing issues with netcdf4 times

I think that it is worth considering making it public and adapting the first few comment lines as the docstring.

I would likely leave the `note that this behaviour...` as a developer comment and not include it in teh docstring in this case